storage/heap/ha_heap.cc

```diff
@@ -854,6 +854,15 @@ heap_prepare_hp_create_info(TABLE *table_arg, bool internal_table,
       {
         fixed_data_size= next_field_pos;
       }
+
+
+      if (field->field_index >= fixed_key_fieldnr)
```

Comment:
Will this work for the case when BLOB is between fields being indexed, like
```sql
CREATE TABLE t1 (a INT, b BLOB, c INT, PRIMARY KEY(a, c)) ENGINE=MEMORY
```

Reply:
Yes this code iterates through all the key parts. The testcase above for the composite keys covers it